packages/apps/backend/src/data-stores/PartStore.ts

```diff
@@ -46,7 +45,19 @@ export class PartStore {
 							const partId = transformers.parts.transformPartId(corePartId)
 
 							if (part) {
-								if (!isEqual(this.parts.get(part._id), part)) this._updatePart(partId, part)
+								const oldPart = this.parts.get(partId)
+								const hasScriptChanged = oldPart && oldPart.scriptContents !== part.scriptContents
```

Comment:
Better to use isEqual for the comparision here, to guard against a future change of scriptContents into an object.